### PR TITLE
Test/msdk 677 refactor entity mocks

### PR DESCRIPTION
**Pull request by monitz87**

### CODE REVIEW

File: src/testUtils/mocks/entities.ts [hunk 1]
```diff
@@ -693,2143 +1470,315 @@ export const mockKnownPermissionGroupModule = (path: string) => (): Record<strin
...
-
-  return securityTokenPermissionsGetAgentsStub;
-}
-
-/**
- * @hidden
- * Retrieve an Authorization Request instance
- */
-export function getAuthorizationRequestInstance(
+export const getAuthorizationRequestInstance = function (
```
Comment (polymath-eric) [inline src/testUtils/mocks/entities.ts]: (style) I think if we are assigning to a variable and exporting that, then these should be arrow functions.

I think its OK to leave as `export function funcName() {` as well, but variable assignment with function keyword doesn't seem idiomatic to me.

File: src/api/entities/SecurityToken/Compliance/__tests__/TrustedClaimIssuers.ts [hunk 2]
```diff
@@ -215,7 +211,7 @@ describe('TrustedClaimIssuers class', () => {
 
       const result = await trustedClaimIssuers.get();
 
-      expect(result).toEqual(expectedTrustedClaimIssuers);
+      expect(result).toEqual(expectedDids.map(did => expect.objectContaining({ did })));
```
[the remark this answers is not in the document]
Comment (polymath-eric) [reply src/api/entities/SecurityToken/Compliance/__tests__/TrustedClaimIssuers.ts:214]: Might be OK to leave. At first I was confused with expect thinking it was an assertion in map, but I saw it enough in this PR to get used to it.

File: src/base/__tests__/Context.ts [hunk 3]
```diff
@@ -622,7 +621,7 @@ describe('Context class', () => {
       expect(result).toEqual(unsubCallback);
       sinon.assert.calledWithExactly(callback, {
         allowance: utilsConversionModule.balanceToBigNumber(allowance),
-        subsidizer: entityMockUtils.getAccountInstance({ address: 'payingKey' }),
+        subsidizer: sinon.match({ address: 'payingKey' }),
```
Comment (prashantasdeveloper) [inline src/base/__tests__/Context.ts:624]:
```suggestion
        subsidizer: expect.objectContaining({ address: 'payingKey' }),
```

File: src/api/entities/SecurityToken/Compliance/__tests__/TrustedClaimIssuers.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (prashantasdeveloper) [reply src/api/entities/SecurityToken/Compliance/__tests__/TrustedClaimIssuers.ts:214]: We can leave it as it is

File: src/api/entities/common/namespaces/__tests__/Authorizations.ts [hunk 4]
```diff
@@ -177,16 +167,19 @@ describe('Authorizations class', () => {
...
-      entityMockUtils.getAuthorizationRequestInstance(authParams);
-
       const result = await authsNamespace.getOne({ id });
 
-      expect(result).toEqual(entityMockUtils.getAuthorizationRequestInstance(authParams));
+      expect(result.authId).toEqual(authId);
+      expect(result.expiry).toBeNull();
+      expect(result.data).toEqual(data);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      expect((result.target as any).did).toEqual(did);
```
[the remark this answers is not in the document]
Comment (prashantasdeveloper) [reply src/api/entities/common/namespaces/__tests__/Authorizations.ts:181]: Nice catch

**Review by prashantasdeveloper — COMMENTED** [review]
Other than nits by Eric, it LGTM. Great job done here Jere

**Review by VictorVicente — COMMENTED** [review]
LGTM too. 
One question, why `cleanup()` is not required any more?

File: src/testUtils/mocks/entities.ts [hunk 5]
```diff
@@ -74,207 +75,165 @@ import {
...
+export type MockCheckpointSchedule = Mocked<CheckpointSchedule>;
+export type MockCorporateAction = Mocked<CorporateAction>;
+export type MockDividendDistribution = Mocked<DividendDistribution>;
+export type MockCustomPermissionGroup = Mocked<CustomPermissionGroup>;
+export type MockKnownPermissionGroup = Mocked<KnownPermissionGroup>;
+
+interface EntityOptions {
   exists?: boolean;
+  isEqual?: boolean;
+  toJson?: any;
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/testUtils/mocks/entities.ts]: Could you open a discussion for this in the middleware channel?

[the remark this answers is not in the document]
Comment (polymath-eric) [reply src/testUtils/mocks/entities.ts]: https://polymathnetwork.slack.com/archives/C026L2FKT25/p1643921067381539

**Comment by monitz87** [comment]
> LGTM too. One question, why `cleanup()` is not required any more?

The cleanup method only "emptied" the mock instance container. Since we're no longer holding an instance in memory per entity, there's nothing to clean up. Every mock entity class is independent and starts fresh, and the `reset` function is enough to get the config back to the initial values

**Review by VictorVicente — APPROVED** [review]

**Review by prashantasdeveloper — APPROVED** [review]

**Pull request merged**